hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -345,7 +355,7 @@ public static <T extends Throwable> T cloneExceptionWithFixedAsyncStackTrace(T o
             return (T) clone;
         }
 
-        return null;
+        return original;
```

Comment:
Seeing null checks based on return type of this method, can we also remove those null checks?
Additionally, when cloning is failed, we are not returning a wrapped exception any more but the original one, this will be a breaking change, better to mention in release docs.